### ISSUE TITLE
Duplicated archived files

### ISSUE THREAD

**Issue by albertgao228** [NLog/NLog#4804]
Hi! Thanks for reporting this bug! 

Please keep / fill in the relevant info from this template so that we can help you as best as possible.

**NLog version**: (e.g. 4.7.6)

**Platform**: .Net 4.8

**Current NLog config** (xml or C#, if relevant)

```xml
<nlog>
  <targets>
  </targets>
<target name="VSLogFile" xsi:type="File" fileName="${logdir}/MyAppVSLogFile.${shortdate}.${processid}.txt" layout="[${longdate}|${logger:shortName=true}|${threadid}|${level}]- ${message} ${exception:format=tostring}" createDirs="true" maxArchiveFiles="30" archiveNumbering="Sequence" archiveAboveSize="100000000" archiveFileName="${logdir}/NewGEMVSLogFile.${shortdate}.${processid}.{#}.zip" enableArchiveFileCompression="true" concurrentWrites="false" />
  <rules>
      <logger name="MyCompany.Business.BusinessObjects.Pending.*" minlevel="Debug" writeTo="VSLogFile" final="true" />

  </rules>
</nlog>

```

- What is the current result?
**We occasionally saw archived files generated with identical content. Recent incident over 30 archived files created within one minute. Before zipping, the file size was 100MB as defined at target.**
- What is the expected result? 
- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)? **We can't enable internal log because the logging is already very heavy.** 
- Please post full exception details (message, stacktrace, inner exceptions) **No exception**
- Are there any workarounds? yes/no **No**
- Is there a version in which it did work? **Current version works 99% time.**
- Can you help us by writing an unit test? **Sorry can't.**

**Comment by albertgao228** [comment]
Below is the message from my teammate:
I downloaded 2 of these files:

NewGEMVSLogFile.2022-02-22.10224.93.zip
NewGEMVSLogFile.2022-02-22.10224.88.zip

Both of them are exactly identical.

1st line :
[2022-02-22 16:29:47.3976|VSLog|58|Info]- <HandleBasketUpdate Received a BasketUpdate message for ClientBasket ID - 642022200057759105 >
Last line:
[2022-02-22 16:56:51.9946|VSLog|58|Info]- </HandleBasketUpdate>

The data is received over 30 mins – but we have multiple zips of the same logs created.

**Comment by snakefoot** [comment]
This issue is a duplicate of this. https://github.com/NLog/NLog/issues/3807 (That is a duplicate of many others).

The issue is this (Using of dynamic `${shortdate}`):

    fileName="${logdir}/MyAppVSLogFile.${shortdate}.${processid}.txt"

Dynamic archive logic doesn't work together with static archive logic:

    archiveFileName="${logdir}/NewGEMVSLogFile.${shortdate}.${processid}.{#}.zip"
    archiveNumbering="Sequence" 

Instead you should change to this:

    fileName="${logdir}/MyAppVSLogFile.${processid}.txt"
    archiveFileName="${logdir}/NewGEMVSLogFile.{#}.zip"
    archiveNumbering="DateAndSequence"
    enableArchiveFileCompression="true"

Alternative stop using zip-file-compression and just use dynamic-filename-archive-logic:

````xml
   <target name="VSLogFile" xsi:type="File" fileName="${logdir}/MyAppVSLogFile.${shortdate}.${processid}.txt" maxArchiveFiles="30" concurrentWrites="false" />
```

See also: [Do not mix "Dynamic FileName Archive Logic" together with "Static FileName Archive Logic"](https://github.com/NLog/NLog/wiki/File-target#dynamic-vs-static-archive-logic)

**Comment by albertgao228** [comment]
Thanks a lot for super quick response. I will adjust our settings accordingly.

**Comment by albertgao228** [comment]
Is there any problem with this:
<target name="VSLogFile" xsi:type="File"
             fileName="${logdir}/MyAppVSLogFile.${shortdate}.${processid}.txt"
             layout="[${longdate}|${logger:shortName=true}|${threadid}|${level}]- ${message} ${exception:format=tostring}"
             archiveAboveSize="100000000"
             maxArchiveFiles="30"
             archiveFileName="${logdir}/MyAppVSLogFile.${shortdate}.${processid}.{#}.txt"
             concurrentWrites="false" />

Without archiveFileName, the sequence number started with processid. So if the log file is file.2022-02-23.38396.txt, the archived file name was file.2022-02-23-38396.38396.txt
I saw ordering was sequence, which is expected.

Thanks.

**Comment by snakefoot** [comment]
Yes you are probably right that the `${processid}` is confusing the organic-archive-logic.

Maybe something like this is needed:

> fileName="${logdir}/MyAppVSLogFile.${shortdate}_${processid}_PID.txt"
